### PR TITLE
Publish 3.0.1 as v3.0.1-draft

### PR DESCRIPTION
**Pull request by bact**
### What this PR do?

Loosely applying steps 5, 7, and 8 in https://github.com/spdx/spdx-spec/pull/950#issuecomment-2096536032 but using `v3.0.1-draft` as a version for now while 3.0.1 is still in development.

- in .github/workflows/publish_v3.yml:
  - Publish `v3.0.1-draft`, using `development/v3.0.1` of `spdx-spec` and `main` of `spdx-model`
  - Update `actions/checkout` to v4 to prevent build warning of Node.js deprecation

- Use the generated `mkdocs-files.yml` from #978

### Notes

After v3.0.1 is released, another PR can be made to publish `v3.0.1`, using `development/v3.0.1` of `spdx-spec` and `3.0.1` of `spdx-model` -- also assign the version `v3.0.1` with two aliases `v3.0.1-draft` and `latest`.
- `run: mike deploy -u -b gh-pages -p v3.0.1 v3.0.1-draft latest`

### CODE REVIEW

File: mkdocs.yml
Comment (zvr) [inline mkdocs.yml:5]:
```suggestion
copyright: DRAFT -- v3.0.1 -- DRAFT
```

**Review by zvr — CHANGES_REQUESTED** [review]

**Review by zvr — APPROVED** [review]

**Comment by bact** [comment]
Have to copy list of model files (from `mkdocs-files.yml`) to `mkdocs.yml`'s `nav` section

**Comment by goneall** [comment]
@puerco - Should we use commits for the actions rather than tags?

**Comment by bact** [comment]
Can change to

```yaml
uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332  #v4.1.7
```

**Comment by goneall** [comment]
> Can change to
> 
> ```yaml
> uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332  #v4.1.7
> ```

Thanks @bact - this should resolve the possible security issue if the tag is changed.

**Comment by bact** [comment]
@zvr `publish_v3.yml` publication workflow is now using the actual list of model entries generated by spec-parser. See lines 56-69.

Note that I have modify the PR validation workflow `validate_pull_request.yml` to do the same (using the generated list of model entries). This is necessary to have this PR pass all the checks.

**Comment by zvr** [comment]
I think the default shown should be the one we're working on, but this can be decided and changed later.
Waiting for another approval to commit this.

**Comment by bact** [comment]
Agree. Let's have this published first and we can start seeing updated contents and formats. Then made any necessary changes afterwards.

**Review by goneall — APPROVED** [review]
LGTM

**Comment by bact** [comment]
Thank you. Content from `development/v3.0.1` is now published at https://spdx.github.io/spdx-spec/v3.0.1-draft/

**Pull request merged**
